### PR TITLE
Add target features for LoongArch

### PR DESCRIPTION
**Pull request by heiher**

### CODE REVIEW

**Review by wesleywiser — APPROVED** [review]

**Comment by heiher** [comment]
> The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/6584899256/job/17890296624) [(plain)](https://github.com/rust-lang-ci/rust/commit/89fc2e0ab9464952fd490f47dfdabbba1296c3ba/checks/17890296624/logs)
> Click to see the possible cause of the failure (guessed by this bot)

Oops! The `ual` feature is missed in LLVM 16, it's added from LLVM 17.

Is there a way to skip testing for this feature on LLVM 16? Or I remove `ual` first and then add it back when time is right.

File: compiler/rustc_codegen_ssa/src/target_features.rs [hunk 1]
```diff
@@ -342,6 +342,18 @@ const CSKY_ALLOWED_FEATURES: &[(&str, Option<Symbol>)] = &[
...
+
+const LOONGARCH_ALLOWED_FEATURES: &[(&str, Option<Symbol>)] = &[
+    // tidy-alphabetical-start
+    ("d", Some(sym::loongarch_target_feature)),
+    ("f", Some(sym::loongarch_target_feature)),
+    ("lasx", Some(sym::loongarch_target_feature)),
+    ("lbt", Some(sym::loongarch_target_feature)),
+    ("lsx", Some(sym::loongarch_target_feature)),
+    ("lvz", Some(sym::loongarch_target_feature)),
+    // tidy-alphabetical-end
```
Comment (heiher) [inline compiler/rustc_codegen_ssa/src/target_features.rs:354]: `ual` removed.

**Comment by heiher** [comment]
@wesleywiser RFR

A min-llvm-version is added to abi compatibility tests to skip unsupported versions, and we will not lose any target features now.

**Review by wesleywiser — APPROVED** [review]

**Comment by wesleywiser** [comment]
That seems fine to me. Earlier versions of LLVM don't have great LoongArch support anyway. 

@bors r+

**Pull request merged**